Dockerfile

```diff
@@ -124,19 +124,24 @@ RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install --no-instal
 
 # The above image is independent of the contents of this package (except
 # for requirements.txt), and is used to form the image for Jenkins
-# testing. We now install the package in a new build stage.
+# testing. We now install the requirements in a new build stage.
 
 FROM build-base as build-py
 
 # Install requirements (already copied to build-base image).
 WORKDIR /tmp/katgpucbf
 RUN pip install -r requirements.txt
 
-# Install the package itself. Using --no-deps ensures that if there are
-# requirements that aren't pinned in requirements.txt, the subsequent
-# pip check will complain.
+#######################################################################
+
+FROM build-base as build-wheel
+
+# Build a wheel for the package. Note that this happens independently
+# of the build-py image.
+WORKDIR /tmp/katgpucbf
 COPY . .
-RUN pip install --no-deps . && pip check
+RUN pip install -c requirements-dev.txt build
```

Comment:
Since you've added build to `requirements-dev.in`/`.txt`, I don't see the reason for it to be on this line here.